vault/provider_test.go

```diff
@@ -387,17 +386,8 @@ func TestAccProviderToken(t *testing.T) {
 	}
 
 	// Clear the config file env var and restore it after the test.
-	origConfigPath, ok := os.LookupEnv(config.ConfigPathEnv)
-	if ok {
-		// A config path env var was set; ensure we restore it.
-		defer func() {
-			err := os.Setenv(config.ConfigPathEnv, origConfigPath)
-			if err != nil {
-				t.Fatal(err)
-			}
-		}()
-	}
-	err = os.Unsetenv(config.ConfigPathEnv)
+	reset, err := tempUnsetenv(config.ConfigPathEnv)
```

Comment:
I ended up using TestAccProviderToken as a template for writing TestAccProviderVaultAddEnv, and in the process of reading them and writing another similar test consolidated some of the setup/tear down refactoring -- if this is not ok I can unwind the changes here